qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/FaissSQIT.java

```diff
@@ -20,10 +20,10 @@
 import org.opensearch.core.xcontent.XContentBuilder;
 import org.opensearch.knn.KNNResult;
 import org.opensearch.knn.common.KNNConstants;
-import org.opensearch.knn.index.KNNMethod;
+import org.opensearch.knn.index.engine.KNNMethod;
 import org.opensearch.knn.index.SpaceType;
 import org.opensearch.knn.index.query.KNNQueryBuilder;
-import org.opensearch.knn.index.util.KNNEngine;
+import org.opensearch.knn.index.engine.KNNEngine;
```

Comment:
Why does it still have to be under index package?

Reply:
I think `org.opensearch.knn.index` is base package for knn index in general. I guess we can move it one more up `org.opensearch.knn.engne` as @shatejas mentioned. What do you think @jmazanec15 ?

Reply:
Thats a good question. Its been mostly convention to have index apply to Lucene index/shard related operations in OpenSearch (see https://github.com/opensearch-project/OpenSearch/tree/main/server/src/main/java/org/opensearch/index). I could move it out, but might be better to consider later as engine is pretty shard specific.

Reply:
I will leave this up to you. 

The concern is majority of code inside index package. For instance, KNNQueryBuilder which isn't shard specific goes under index

Reply:
Yes, that isnt great. I think this is probably something that can be addressed later in separate PR refactoring index package.